File: app/src/main/java/nl/koenhabets/yahtzeescore/PlayerItem.java
package nl.koenhabets.yahtzeescore;

public class PlayerItem implements Comparable<PlayerItem> {
    private String name;
    private Integer score;
    private long lastUpdate;
    private boolean visible;
    private boolean local;

    public PlayerItem(String name, Integer score, long lastUpdate, boolean visible, boolean local) {
        this.name = name;
        this.score = score;
        this.lastUpdate = lastUpdate;
        this.visible = visible;
        this.local = local;
    }

    public String getName() {
        return name;
    }

    public Integer getScore() {
        return score;
    }

    public long getLastUpdate() {
        return lastUpdate;
    }

    public boolean isVisible() {
        return visible;
    }

    public void setVisible (boolean visible) {
        this.visible = visible;
    }

    @Override
    public int compareTo(PlayerItem p) {
        return p.getScore().compareTo(this.getScore());
    }

    public boolean isLocal() {
        return local;
    }
}

File: app/src/main/java/nl/koenhabets/yahtzeescore/model/PlayerItem.kt
package nl.koenhabets.yahtzeescore.model

import kotlinx.serialization.json.JsonObject

data class PlayerItem(
    val id: String,
    var name: String?,
    var score: Int?,
    var fullScore: JsonObject?,
    var lastUpdate: Long,
    val isLocal: Boolean,
    var game: String?
) : Comparable<PlayerItem> {
    override fun compareTo(other: PlayerItem): Int {
        other.score?.let {
            this.score?.let { scoreThis ->
                return it.compareTo(scoreThis)
            }
        }
        return 0
    }

    /*    val scoreCount: Int
            get() {
                var count = 0
                for (i in 0..13) {
                    var d = i
                    if (d > 6) {
                        d = i + 15
                    }
                    try {
                        if (fullScore.has(d.toString())) {
                            if (fullScore.getString(d.toString()) != "") {
                                count++
                            }
                        }
                    } catch (e: JSONException) {
                        e.printStackTrace()
                    }
                }
                return count
            }*/
}